autotest/gdrivers/tiledb_write.py

```diff
@@ -122,47 +144,52 @@ def test_tiledb_write_attributes():
     gdal.GetDriverByName('GTiff').Create('/vsimem/temp1.tif', w, h, num_bands, gdal.GDT_Int32)
     gdal.GetDriverByName('GTiff').Create('/vsimem/temp2.tif', w, h, num_bands, gdal.GDT_Float32)
 
-    options = [
-        'TILEDB_ATTRIBUTE=%s' % ('/vsimem/temp1.tif'),
-        'TILEDB_ATTRIBUTE=%s' % ('/vsimem/temp2.tif')
-    ]
-
-    new_ds = gdaltest.tiledb_drv.CreateCopy('tmp/tiledb_rgb_atts', src_ds, options=options)
-    assert new_ds is not None
-    assert new_ds.RasterXSize == src_ds.RasterXSize
-    assert new_ds.RasterYSize == src_ds.RasterYSize
-    assert new_ds.RasterCount == src_ds.RasterCount
-
-    new_ds = None
-
-    # check we can open the attributes with the band as well as the pixel values
-    att1_ds = gdal.OpenEx('tmp/tiledb_rgb_atts', open_options=['TILEDB_ATTRIBUTE=temp1'])
-    att2_ds = gdal.OpenEx('tmp/tiledb_rgb_atts', open_options=['TILEDB_ATTRIBUTE=temp2'])
-    val_ds = gdal.Open('tmp/tiledb_rgb_atts')
-
-    meta = val_ds.GetMetadata('IMAGE_STRUCTURE')
-    assert 'TILEDB_ATTRIBUTE_1' in meta
-    assert meta['TILEDB_ATTRIBUTE_1'] == 'temp1'
-    assert 'TILEDB_ATTRIBUTE_2' in meta
-    assert meta['TILEDB_ATTRIBUTE_2'] == 'temp2'
-
-    assert att1_ds is not None
-    assert att2_ds is not None
-    assert val_ds is not None
-
-    assert att1_ds.GetRasterBand(1).DataType == gdal.GDT_Int32
-    assert att2_ds.GetRasterBand(1).DataType == gdal.GDT_Float32
-    assert val_ds.RasterCount == src_ds.RasterCount
-    assert val_ds.GetRasterBand(1).DataType == src_ds.GetRasterBand(1).DataType
+    for mode in [ 'BAND', 'PIXEL' ]:
```

Comment:
a more pytest'ic way would be to use pytest.parametrize (see autotest/gdrivers/fits.py for an example)